[[ Bug 22946 ]] Update the engine to compile with Xcode 12

livecodepanos opened the pull request:
This patch addresses various errors when building with Xcode 12.

It needs https://github.com/livecode/livecode/pull/7449

runrevmark commented on engine/src/mbliphoneextra.mm:
```diff
@@ -711,13 +711,8 @@ bool MCSystemGetCurrentLocale(MCStringRef& r_current_locale)
-	
-    return MCStringCreateWithCFStringRef((CFStringRef)t_identifier, r_identifier);
+    // Deprecated in LC 6.0
+    return false;
```
This will cause an error to be thrown - we want to return empty:
```
{
    r_identifier = MCValueRetain(kMCEmptyString);
    return true;
}
```

runrevmark replied:
This also requires a distinct bug filed in LC and a release note added for it - e.g. `The long deprecated mobileGetSystemIdentifier command now returns empty.`

(I think thats the name of the command...)

runrevmark reviewed (changes requested):
Can you file a BZ issue (in addition to the one for the change to GetSystemIdentifier) about 'update the engine to compile with Xcode 12). [ This aspect doesn't require a release note - but its useful to record the change as an issue in BZ ].

Can you also separate the commits so all tweaks to config/source to make things compile is one commit; and the change to mobileGetSystemIdentifier (or is it iphoneGetSystemIdentifier?) in another.

runrevmark reviewed (changes requested):
@livecodepanos As mentioned in the previous review, please file separate BZ's about both the xcode 12 update and the change to the iphoneSystemIdentifier - and add a release note for the latter. It is probably worth splitting into two PRs as well (as discussed offline).

livecodepanos commented:
@runrevmark 

BZ reports:

1. Update the engine to compile with Xcode 12: https://quality.livecode.com/show_bug.cgi?id=22946
2. Change the behavior of deprecated `iphoneSystemIdentifier`: https://quality.livecode.com/show_bug.cgi?id=22947

 PR for (2): https://github.com/livecode/livecode/pull/7456

livecodepanos committed:
[[ Bug 22946 ]] Update the engine to compile with Xcode 12

Changes:

1. Remove unused deprecated Framework that caused compile error on iOS 14  …
Fixes compile error:
- error: expected '('
MP_DEPRECATED_BEGIN("Use AVPlayerViewController in AVKit.", ios(3.2, 9.0))
^- error: expected 'push' or 'pop' after '#pragma clang attribute'

2. Fix error "implicit declaration of function X' is invalid in C99" by adding a proper flag in ios.gypi

3. Removed notifications code for iOS < 8, since the oldest iOS version we support is 8.0.
Fixes error:
error: no matching function for call to 'objc_msgSend'
objc_msgSend([UIApplication sharedApplication], sel_getUid("registerForRemoteNotificationTypes:"), t_allowed_notifications);

4. Added inclusion of missing header.
Fixes error:
/Users/administrator/Vulcan/slaves/build-mac-4-1/slave/try-community-universal-ios-iphoneos14_0/checkout/toolchain/lc-compile-ffi-java/src/main.c:73:17: error: implicit declaration of function 'SetOutputFile' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                SetOutputFile(argv[++argi]);
                ^
/Users/administrator/Vulcan/slaves/build-mac-4-1/slave/try-community-universal-ios-iphoneos14_0/checkout/toolchain/lc-compile-ffi-java/src/main.c:78:17: error: implicit declaration of function 'SetOutputLCBFile' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                SetOutputLCBFile(argv[++argi]);
                ^
/Users/administrator/Vulcan/slaves/build-mac-4-1/slave/try-community-universal-ios-iphoneos14_0/checkout/toolchain/lc-compile-ffi-java/src/main.c:83:17: error: implicit declaration of function 'SetOutputLCBModuleName' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                SetOutputLCBModuleName(argv[++argi]);

runrevmark reviewed (changes requested):
Can you tweak the commit message title so that it is:
```
[[ Bug 22946 ]] Update the engine to compile with Xcode 12
```

livecodepanos commented:
@runrevmark Done :)

runrevmark commented:
@livecode-vulcan review ok 54489a1

livecode-vulcan commented:
💙 review by @runrevmark ok 54489a1501b4a4df362b6d029685ba792dbd6e0a

livecode-vulcan commented:
😎 test success 54489a1501b4a4df362b6d029685ba792dbd6e0a

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/898)
* try-community-armv7-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv7-android-ndk16r15/builds/307)
* try-community-arm64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-arm64-android-ndk16r15/builds/307)
* try-community-x86-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86-android-ndk16r15/builds/307)
* try-community-x86_64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86_64-android-ndk16r15/builds/307)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/2424)
* try-community-universal-ios-iphoneos14.1: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos14.1/builds/1)
* try-community-universal-ios-iphoneos13.5: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos13.5/builds/77)
* try-community-universal-ios-iphonesimulator14.1: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator14.1/builds/1)
* try-community-universal-ios-iphonesimulator13.5: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator13.5/builds/73)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/2451)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/2440)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/2439)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/2474)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/2414)

livecodepanos commented:
Trying again with Xcode 12.x:

@livecode-vulcan retry 54489a1

livecode-vulcan commented:
😎 test success 54489a1501b4a4df362b6d029685ba792dbd6e0a

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/899)
* try-community-armv7-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv7-android-ndk16r15/builds/308)
* try-community-arm64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-arm64-android-ndk16r15/builds/308)
* try-community-x86-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86-android-ndk16r15/builds/308)
* try-community-x86_64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86_64-android-ndk16r15/builds/308)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/2425)
* try-community-universal-ios-iphoneos14.1: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos14.1/builds/2)
* try-community-universal-ios-iphoneos13.5: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos13.5/builds/78)
* try-community-universal-ios-iphonesimulator14.1: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator14.1/builds/2)
* try-community-universal-ios-iphonesimulator13.5: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator13.5/builds/74)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/2452)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/2441)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/2440)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/2475)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/2415)
